File: wiz-shop-java/src/main/java/com/example/wizshopjava/domain/member/repository/MemberRepository.java
package com.example.wizshopjava.domain.member.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.example.wizshopjava.domain.member.entity.Member;

@Repository
public interface MemberRepository extends JpaRepository<Member, Long> {

	boolean existsByEmail(String email);
	Optional<Member> findByEmail(String email);
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/domain/member/repository/MemberRepository.kt
package com.example.wizeventmall.domain.member.repository

import com.example.wizeventmall.domain.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MemberRepository : JpaRepository<Member, Long> {

    fun existsByEmail(email: String): Boolean
    fun findByEmail(email: String): Member?
}